pkg/syslog/rfc5424/reader.go

```diff
@@ -0,0 +1,80 @@
+package rfc5424
+
+import (
+	"bufio"
+	"bytes"
+	"strconv"
+	"time"
+)
+
+type Reader struct {
+	s scanner
+}
+
+func NewReader(s *bufio.Scanner) *Reader {
+	return &Reader{scanner{s}}
+}
+
+func (r *Reader) ReadMessage() (*Message, error) {
+	parts, err := r.s.next()
+	if err != nil {
+		return nil, err
+	}
+
+	priority, err := parseDecimal(parts[0])
+	if err != nil {
+		return nil, err
+	}
+
+	version, err := parseDecimal(parts[1])
+	if err != nil {
+		return nil, err
+	}
+
+	timestamp, err := parseTimestamp(parts[2])
+	if err != nil {
+		return nil, err
+	}
+
+	m := &Message{
+		Header: Header{
+			Priority:  priority,
+			Version:   version,
+			Timestamp: timestamp,
+			Hostname:  parseMaybeNil(parts[3]),
+			AppName:   parseMaybeNil(parts[4]),
+			ProcID:    parseMaybeNil(parts[5]),
+			MsgID:     parseMaybeNil(parts[6]),
+		},
+		StructuredData: parseMaybeNil(parts[7]),
```

Comment:
I don't think this works, because structured data contains spaces. May be better to explicitly disallow it until we can support it properly (if we ever even need to).